scavenge/tutorial/04-messages.md

```diff
@@ -0,0 +1,68 @@
+---
+order: 4
+---
+
+# Messages
+
+Messages are a great place to start when building a module because they define the actions that your application can make. Think of all the scenarios where a user would be able to update the state of the application in any way. These should be boiled down into basic interactions, similar to **CRUD** (Create, Read, Update, Delete).
+
+The Scavenge module will have 3 messages:
+
+* Create scavenge
+* Commit solution
+* Reveal solution
+
+## Create Scavenge Message
+
+Create scavenge message should contain all the necessary information when creating a scavenge:
+
+* Description - what is the question to be solved or description of the challenge.
+* Solution hash - the scrambled solution.
+* Reward - this is the bounty that is awarded to whoever submits the answer first.
+
+Use the `starport scaffold message` command to scaffold a new Cosmos SDK message for your module. The command accepts message name as the first argument and a list of fields. By default, a message is scaffolded in a module with a name that matches the name of the project, in our case `scavenge` (this behaviour can be overwritten by using a flag).
+
+```
+starport scaffold message create-scavenge solutionHash description reward
```

Comment:
I see `solutionHash` as camelCase here, juding by PR https://github.com/cosmos/sdk-tutorials/pull/708/files I think we are trying to avoid camelCase and use dash-case rather?